hubclient/client.go

```diff
@@ -181,9 +181,12 @@ func (c *Client) processResponse(resp *http.Response, result interface{}, expect
 
 func (c *Client) HttpGetString(url string, result *string, expectedStatusCode []int, mimetypes ...string) (error, int) {
 	err, response := c.httpGet(url, nil, expectedStatusCode, mimetypes...)
+
+	statusCode := getResponseStatus(response)
```

Comment:
I think you are right - there are "errors" that could still have a body.
I guess we just need a check for response to be nil before we try to use it (dereference), and that's pretty much it.
---
The more I look at this function, the less I like it. We need to return 3 items from here actually - string, err, and code (Or wrap HTTP code into error, but I am not sure how many places will choke because it will not be nil anymore). Passing pointer for the result (and btw, not checking it for nil) is not exactly Go-like style, where multiple return values are solving this exact problem.